addons/web/static/src/core/assets.js

```diff
@@ -65,109 +68,166 @@ export const loadCSS = memoize(function loadCSS(url) {
         });
     });
 });
-/**
- * Loads the qweb templates from a given bundle name.
- * TODO: merge this into loadBundleDefinition?
- *
- * @param {string} bundle the name of the bundle as declared in the manifest.
- * @returns {Promise<XMLDocument|"">} A Promise of an XML document containing
- *      the owl templates or an empty string if the bundle has none.
- */
-export const fetchAndProcessTemplates = memoize(async function fetchAndProcessTemplates(bundle) {
-    // TODO: quid of the "unique" in the URL? We can't have one cache_hash
-    // for each and every bundle I'm guessing.
-    const bundleURL = `/web/webclient/qweb/${Date.now()}?bundle=${bundle}`;
-    const templates = await (await browser.fetch(bundleURL)).text();
-    if (!templates) {
-        return "";
-    }
-    return processTemplates(templates);
-});
+
+export const templates = new DOMParser().parseFromString("<odoo/>", "text/xml");
 
 /**
- * Loads the content definition of a bundle.
+ * Loads the given xml template.
  *
- * @param {string} name the bundleName of the bundle as declared in the manifest.
- * @returns {Promise<BundleInfo[]>} A promise of the content definition of the bundle
+ * @param {string} xml the string defining the templates
+ * @param {App} [app=assets.defaultApp] optional owl App instance
+ * @returns {Promise<true>} resolved when the template xml has been loaded
  */
-const loadBundleDefinition = memoize(async function (bundleName) {
-    const request = await browser.fetch(`/web/bundle/${bundleName}`);
-    return await request.json();
-});
+ assets.loadXML = function loadXML (xml, app=assets.defaultApp) {
+    const doc = new DOMParser().parseFromString(xml, "text/xml");
+    if (doc.querySelector('parsererror')) {
+        throw doc.querySelector('parsererror div').textContent.split(':')[0];
+    }
 
-const bundlesCache = {};
+    for (const element of doc.querySelectorAll("templates > [t-name][owl], templates > [t-extend][owl]")) {
```

Comment:
Are there templates that are t-extend and owl at the same time? How does that even work?

Reply:
no `t-extend` and `owl` it's a mistake. But they are some template with `t-inherit` + `t-extend`